### DISCUSSION TITLE
Annotazioni automatiche - richiesta nuovi campi nella risposta

### DISCUSSION THREAD

**Discussion by siscomanpr** [italia/ansc#836] [Miglioramenti] [answered]
Buongiorno,
parlando di Annotazioni automatiche che vengono generate dal sistema in seguito ad un evento, riteniamo possa essere utile ottenere tra i campi della response (nel modelEvento relativo all'annotazione), anche il numero di formula (da formulario del Decreto Ministeriale 5 aprile 2002) utilizzato per comporre il testo dell'annotazione. Oppure in alternativa, se è meno problematico, nella response della ricerca delle notifiche di tipo Annotazione.

La richiesta nasce dal fatto che ci sono dei casi in cui a fronte di un evento, vengono generate 2 annotazioni automatiche diverse, ma che è impossibile andare a distinguere in modo automatico per andare ad importarle nella banca dati locale del nostro gestionale senza l'intervento manuale dell'operatore. 

Faccio un esempio:
Evento nascita con caso d'uso 1331 (trascrizione decreto cambio cognome).
Supponiamo che il soggetto abbia un atto di nascita primario cartaceo e che sia coniugato o vedovo: 
Il sistema genera 2 annotazioni automatiche da inviare via PEC con il campo idTipoEvento=1.
Una per l'atto di nascita del soggetto e una per l'atto di nascita del coniuge.
Queste 2 annotazioni, a parte i campi della composizioneCompleta e degli identificativi, per il resto sono uguali (anche il campo idSoggettoRif è lo stesso dell'intestatario dell'atto).

Un esempio è l'evento con idAnsc 2024-1312911-48-002137 sull'ambiente di pre-produzione di Serravalle Sesia, e le 2 annotazioni generate hanno i seguenti idAnsc: 2024-1312921-00000-002137 e 2024-1312922-00000-002137

Se ci fosse un indicatore, o appunto il numero della formula utilizzata per comporre il testo dell'annotazione, riusciremmo a contestualizzare correttamente le 2 annotazioni.

Sempre parlando di Annotazioni automatiche, quelle invece con idcanalenotifica=0 (da registrare su atti primari già digitali), sarebbe molto utile avere l'ID dell'atto primario su cui il sistema centrale andrebbe poi a registrare l'annotazione una volta approvata.
Anche questa miglioria ci consentirebbe di allineare la banca dati locale del gestionale in maniera più automatizzata e puntuale.

Grazie per l'attenzione.

Mauro Maccagno
Siscom SPA

**Comment by mttfranci** [comment, marked as answer]
Salve, valuteremo con il DTD se è possibile pianificare la modifica richiesta.

Saluti

Assistenza ANSC

**Comment by asqueglia** [comment]
Come da roadmap https://github.com/italia/ansc/blob/main/docs/Roadmap.md l'intervento sarà previsto ad ottobre a meno di diverse esigenze.

**Comment by asqueglia** [comment]
Con il rilascio della versione 1.34.0 la segnalazione è stata risolta
